bfvmm/integration/arch/intel_x64/test_all.sh

```diff
@@ -0,0 +1,370 @@
+          $rbx_count -eq $rcx_count && $rcx_count -eq $rdx_count  &&
+          $rdx_count -eq $cpu_count ]];
+    then
+        return 0
+    fi
+
+    echo ""
+    echo_fail "observed $record_count records; expected $cpu_count"
+    echo_fail "observed $leaf_count leaf; expected $cpu_count"
+    echo_fail "observed $subleaf_count subleaf; expected $cpu_count"
+    echo_fail "observed $rax_count rax; expected $cpu_count"
+    echo_fail "observed $rbx_count rbx; expected $cpu_count"
+    echo_fail "observed $rcx_count rcx; expected $cpu_count"
+    echo_fail "observed $rdx_count rdx; expected $cpu_count"
+    echo ""
+
+    exit 1
+}
+
+check_io()
+{
+    local record_count=$(grep ": record" serial.out | wc -l)
+    local port_count=$(grep "\- port_number" serial.out | wc -l)
+    local size_count=$(grep "\- size_of_access" serial.out | wc -l)
+    local direction_count=$(grep "\- direction_of_access" serial.out | wc -l)
+    local address_count=$(grep "\- address " serial.out | wc -l)
+    local val_count=$(grep "\- val" serial.out | wc -l)
+
+    if [[ $record_count -eq $port_count && $port_count -eq $size_count &&
+          $size_count -eq $direction_count && $direction_count -eq $address_count &&
+          $address_count -eq $val_count ]];
+    then
+        return 0
+    fi
+
+    echo ""
+    echo_fail "observed $record_count records; expected $record_count"
+    echo_fail "observed $port_count port; expected $record_count"
+    echo_fail "observed $size_count size; expected $record_count"
+    echo_fail "observed $direction_count direction; expected $record_count"
+    echo_fail "observed $address_count address; expected $record_count"
+    echo_fail "observed $val_count val; expected $record_count"
+    echo ""
+
+    exit 1
+}
+
+check_test()
+{
+    local vmm=$1
+
+    case $vmm in
+        *_x64_drs*) check_dr;;
+        *_x64_cpuid*) check_cpuid;;
+        *_x64_control_register*) check_cr;;
+        *_x64_io_instruction*) check_io;;
+        *_x64_monitor_trap*) check_monitor_trap;;
+        *_x64_*msr*) check_msr;;
+        *_x64_vic*) check_vic;;
+        *_x64_vpid*) check_vpid;;
+        *)
+            echo ""
+            echo -ne "$bold_yellow"
+            echo -e  "ALERT:$reset check not supported: $vmm"
+            ;;
+    esac
+}
+
+#
+# Test initializers
+#
+
+init_vic()
+{
+    sudo dmesg -C
+}
+
+init_test()
+{
+    local vmm=$1
+
+    case $vmm in
+        *_x64_vic*) init_vic;;
+        *) ;;
+    esac
+}
+
+#
+# Prettify the test names
+#
+
+parse_test_name()
+{
+    name=$(basename $1 | sed 's|eapis_integration_intel_x64_\(.*\)_static|\1|')
+    echo "$name"
+}
+
+#
+# Global variables
+#
+
+case $LANG in
+    en_US\.UTF-8)
+        pass_mark='\u2713'
+        fail_mark='\u2718'
+        ;;
+    *)
+        pass_mark='*'
+        fail_mark='x'
+        ;;
+esac
+
+pass_mark="$bold_green$pass_mark$reset"
+fail_mark="$bold_red$fail_mark$reset"
+
+build_dir=$HOME/bareflank/build-eapis
```

Comment:
Can we move the global variables to the top of this file so its easy to find them and edit the paths for your local environment? Or maybe alternately the script could accept build_dir as an argument, and then default to the path defined here?